Sources/Datadog/Logging/LogOutputs/LogFileOutput.swift

```diff
@@ -6,18 +6,16 @@
 
 import Foundation
 
-/// `LogOutput` which saves logs to file.
+/// `LogOutput` writing logs to file.
 internal struct LogFileOutput: LogOutput {
-    let logBuilder: LogBuilder
     let fileWriter: Writer
     /// Integration with RUM Errors.
     let rumErrorsIntegration: LoggingWithRUMErrorsIntegration?
 
-    func writeLogWith(level: LogLevel, message: String, error: DDError?, date: Date, attributes: LogAttributes, tags: Set<String>) {
-        let log = logBuilder.createLogWith(level: level, message: message, error: error, date: date, attributes: attributes, tags: tags)
+    func write(log: Log) {
         fileWriter.write(value: log)
 
-        if level.rawValue >= LogLevel.error.rawValue {
+        if log.status == .error || log.status == .critical {
             rumErrorsIntegration?.addError(for: log)
```

Comment:
when i worked on this part recently, i didn't fully understand why passing logs to RUM is `LogFileOutput`s responsibility and why doesn't `LogFileOutputTests` have it as a test case?

is it a product requirement that RUM integration works only with file output and not with console output?

Reply:
This is tested using behaviour test defined in `LoggerTests`:
```swift
func testWhenSendingErrorOrCriticalLogs_itCreatesRUMErrorForCurrentView()
```
Thanks to this, we can move this code freely without touching its test. `LogFileOutput` is just one of the places it could be defined at (it could be done in `Logger` as well).

Reply:
> is it a product requirement that RUM integration works only with file output and not with console output?

The reasoning here was: if the user doesn't want logs to be send to Datadog, he/she won't want RUM Errors for that logs to be send as well. It wasn't a product requirement.

Reply:
the single case in `LogFileOutputTests` is already covered by each case in `LoggerTests` but it is still there, this is confusing to me: why do we test `testItWritesLogToFileAsJSON` but not `testItSendsLogsToRUM` in the same place?

wouldn't it be easier to manage (eg: catching missing cases) if we test all `LogFileOutput` does in `LogFileOutputTests`?

Reply:
Discussed on Slack 🙂👌.